src/usb_moded-network.c

```diff
@@ -299,18 +300,16 @@ static int resolv_conf_dns(struct ipforward_data *ipforward)
 
 static int checklink(void)
 {
-   char *dest = malloc( 32 * sizeof(char)) ;
-   int ret = 0;
-
-   readlink("/etc/udhcpd.conf", dest, 32);
-   if(dest != NULL)
-   {
-	strcpy(&dest[31], "\0");
-	ret = (strcmp(dest, UDHCP_CONFIG_PATH));
-   }
-   free(dest);
-
-   return(ret);
+  int ret = -1;
+  const size_t bufsiz = 32;
+  char *dest = malloc(bufsiz);
```

Comment:
@spiiroin: `UDHCP_CONFIG_PATH` (sorry, I meant `UDHCP_CONFIG_PATH`, not `UDHCP_CONFIG_LINK`) is `char[]`, so `sizeof(UDHCP_CONFIG_PATH)` is 27 (length + null terminator)